File: src/main/scala/dregex/impl/AsciiHelper.scala
package dregex.impl

object AsciiHelper {

  def isUpper(ch: Char): Boolean = {
    ch >= 'A' && ch <= 'Z'
  }

  def toLower(ch: Char): Char = {
    if (isUpper(ch))
      (ch + 0x20).toChar
    else
      ch
  }

}

File: src/main/scala/dregex/impl/AsciiHelper.java
package dregex.impl;

public class AsciiHelper {

  public static boolean isUpper(char ch) {
      return ch >= 'A' && ch <= 'Z';
  }

  public static char toLower(char ch) {
      return isUpper(ch) ? (char) (ch + 0x20) : ch;
  }
}
